### PR TITLE
follow proxy settings

### LINKED ISSUE #139: Checkout needs to following http proxy env
Thank you 🙇‍♀ for wanting to create an issue in this repository. Before you do, please ensure you are filing the issue in the right place. Issues should only be opened on if the issue **relates to code in this repository**.

* If you have found a security issue [please submit it here](https://hackerone.com/github)
* If you have questions about writing workflows or action files, then please [visit the GitHub Community Forum's Actions Board](https://github.community/t5/GitHub-Actions/bd-p/actions)
* If you are having an issue or question about GitHub Actions then please [contact customer support](https://help.github.com/en/articles/about-github-actions#contacting-support)

If your issue is relevant to this repository, please delete this text and continue to create this issue. Thank you in advance.

### PR DESCRIPTION
**Pull request by ericsciple**
fixes #139

### CODE REVIEW

File: __test__/configure-proxy.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+echo "" >> /etc/hosts
+echo "127.0.0.0 github.com api.github.com" >> /etc/hosts
```
Comment (ericsciple) [inline __test__/configure-proxy.sh]: playing around with a couple other ideas rather than hosts file...

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "checkout",
-  "version": "2.0.1",
+  "version": "2.0.2",
```
Comment (ericsciple) [inline package.json]: bump to 2.1.0

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -99,3 +99,42 @@ jobs:
           path: basic
       - name: Verify basic
         run: __test__/verify-basic.sh --archive
+
+  test-proxy:
```
Comment (ericsciple) [inline .github/workflows/test.yml]: @TingluoHuang i played a bit more with your idea. i think you will like this :)

File: src/git-source-provider.ts [hunk 4]
```diff
@@ -91,11 +95,13 @@ export async function getSource(settings: ISourceSettings): Promise<void> {
       )
     }
 
-    // Remove possible previous extraheader
+    // Remove possible previous proxy and extraheader
+    await removeGitConfig(git, proxyConfigKey)
```
Comment (ericsciple) [inline src/git-source-provider.ts]: remove this since git follows https_proxy and no_proxy

File: package.json [hunk 5]
```diff
@@ -31,7 +31,7 @@
   "dependencies": {
     "@actions/core": "^1.1.3",
     "@actions/exec": "^1.0.1",
-    "@actions/github": "^2.0.0",
+    "@actions/github": "^2.0.2",
```
Comment (i5golfzaa) [inline package.json]: 0

**Pull request merged**